### PR TITLE
Deprecate `@pytest.mark.style(...)`.

### PR DESCRIPTION
**Pull request by anntzer**
`@mpl.style.context` can be used in the same way (and is just as
greppable), but doesn't require any support from the pytest machinery.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/tests/test_patches.py [hunk 1]
```diff
@@ -197,7 +198,7 @@ def test_patch_alpha_override():
     ax.set_ylim([-1, 2])
 
 
-@pytest.mark.style('default')
+@mpl.style.context('default')
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_patches.py:201]: There's an `mstyle` import here already, that you either could use, or drop and change the other usage to match.

**Review by QuLogic — APPROVED** [review]

File: lib/matplotlib/tests/test_patches.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/tests/test_patches.py:201]: mpl.style it is, then.

**Review by QuLogic — CHANGES_REQUESTED** [review]
Tests say that you missed one usage, _and_ that the deprecation is broken.

**Comment by anntzer** [comment]
Thanks, should be fixed now (crossed PR and all...).

**Review by QuLogic — APPROVED** [review]

**Pull request merged**